deno-installer.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+set -u
+
+main() {
+
+  need_cmd downloader --check
+  need_cmd uname
+  need_cmd mktemp
+  need_cmd chmod
+  need_cmd mkdir
+  need_cmd rm
+  need_cmd rmdir
+  need_cmd jq
+
+  get_architecture || return 1
+
+  local _release_name=deno_"$RETVAL"_x64
+
+  get_release_url "$_release_name"
+
+  local _ext=".gz"
+  case "$RETVAL" in
+  *win*)
+      _ext=".zip"
+      ;;
+  esac
+
+  local _dir="$(mktemp -d 2>/dev/null || ensure mktemp -d -t deno_installer)"
+  local _file="${_dir}/deno_${RETVAL}_x64${_ext}"
+
+  ensure downloader $RELEASE_URL $_file
+
+  local _deno_home="${HOME}/deno"
+  local _deno_bin="$_deno_home/deno"
+
+  ensure mkdir -p "$_deno_home"
+
+  gunzip -c "$_file" > "$_deno_bin"
+
+  ensure chmod u+x "$_deno_bin"
+
+  ignore rm "$_file"
+  ignore rmdir "$_dir"
+
+}
+
+# This wraps curl or wget. Try curl first, if not installed,
+# use wget instead.
+downloader() {
+    if check_cmd curl
+    then _dld=curl
+    elif check_cmd wget
+    then _dld=wget
+    else _dld='curl or wget' # to be used in error message of need_cmd
+    fi
+
+    if [ "$1" = --check ]
+    then need_cmd "$_dld"
+    elif [ "$_dld" = curl ]
+    then curl -sSfL "$1" -o "$2"
+    elif [ "$_dld" = wget ]
+    then wget "$1" -O "$2"
+    else err "Unknown downloader"   # should not reach here
+    fi
+}
+
+# the only way I found to get the URL to the latest release from github
+# unfortunatelly it required `jq` to parse JSON output
```

Comment:
Good start but I don't want to depend on jq... but I wouldn't mind if you used node or python.

Reply:
Just for figuring out the latest release? Yeah either will do.

Reply:
It's just jq isn't commonly installed - but python or node are.

Reply:
Agreed! Ok. I'll make it so :)

Reply:
@ry Ok. So no more `jq` but I don't know if what I have is better :) its a small python inline script that finds the latest release url. I can extract it if this is ugly or bad.